### PR TITLE
OFW 8.1.0 support

### PR DESCRIPTION
**Pull request by moebius2033**
Added OFW 8.1.0 support, bump hekate version to 4.10.2

### CODE REVIEW

**Comment by fennectech** [comment]
Just compiled this  Nice but it doesnt work   Likely because of issues with emummc.kip being held in the fss0 with atmo

**Comment by moebius2033** [comment]
Yeah, I still haven't solved that one issue. My workaround has been to simply use a payload (specifically fusee-primary) inside of hekate_ipl.ini rather than try to immediately launch into emummc. Likewise, if a person isn't using emummc, it does as expected; loading sept, and in turn allowing homebrew. As it's been said before, emummc is still beta, and so not something I'm entirely aiming for at the moment.

File: bootloader/frontend/fe_info.c [hunk 1]
```diff
@@ -372,6 +372,12 @@ void print_tsec_key()
 		tsec_ctxt.size = 0xF00;
 	else if (pkg1_id->kb == KB_FIRMWARE_VERSION_620)
 		tsec_ctxt.size = 0x2900;
+	else if (pkg1_id->kb == KB_FIRMWARE_VERSION_810)
+	{
+		tsec_ctxt.size = 0x3300;
+		// Exit after TSEC key generation.
+		*((vu16 *)((u32)tsec_ctxt.fw + 0x2DB5)) = 0x02F8;
```
Comment (CTCaer) [inline bootloader/frontend/fe_info.c]: Did you used that randomly without checking the fuc5 code?

File: bootloader/hos/pkg2.c [hunk 2]
```diff
@@ -592,6 +592,49 @@ static kip1_patchset_t _fs_patches_800_exfat[] =
...
+};
+
+static kip1_patchset_t _fs_patches_810_exfat[] =
+{
+	{ "nosigchk", _fs_nosigchk_810_exfat },
+	{ "nogc",     _fs_nogc_810_exfat },
+	{ NULL, NULL }
+};
+
+
```
Comment (CTCaer) [inline bootloader/hos/pkg2.c]: All the above are unneeded.
8.1.0 has the same offsets with 8.0.0..

File: bootloader/hos/pkg2.c [hunk 3]
```diff
@@ -620,7 +663,9 @@ static kip1_id_t _kip_ids[] =
 	{ "FS", "\x2A\xDB\xE9\x7E\x9B\x5F\x41\x77", _fs_patches_700 }, 		 // FS 7.0.0
 	{ "FS", "\x2C\xCE\x65\x9C\xEC\x53\x6A\x8E", _fs_patches_700_exfat }, // FS 7.0.0 exfat
 	{ "FS", "\xB2\xF5\x17\x6B\x35\x48\x36\x4D", _fs_patches_800 }, 		 // FS 8.0.0
-	{ "FS", "\xDB\xD9\x41\xC0\xC5\x3C\x52\xCC", _fs_patches_800_exfat }  // FS 8.0.0 exfat
+	{ "FS", "\xDB\xD9\x41\xC0\xC5\x3C\x52\xCC", _fs_patches_800_exfat }, // FS 8.0.0 exfat
+	{ "FS", "\x6B\x09\xB6\x7B\x29\xC0\x20\x24", _fs_patches_810 },       // FS 8.1.0
+    { "FS", "\xB4\xCA\xE1\xF2\x49\x65\xD9\x2E", _fs_patches_810_exfat }  // FS 8.1.0 exfat
```
Comment (CTCaer) [inline bootloader/hos/pkg2.c]: Whitespace, but whatever. The comments were broken from before.

File: bootloader/hos/secmon_exo.c [hunk 4]
```diff
@@ -109,8 +109,8 @@ void config_exosphere(const char *id, u32 kb, void *warmboot, bool stock)
 		break;
 	default:
 		exoFwNo = kb + 1;
-		if (!strcmp(id, "20190314172056"))
-			exoFwNo++; // ATM_TARGET_FW_800.
+		if (!strcmp(id, "20190531152432"))
+			exoFwNo++; // ATM_TARGET_FW_810.
```
Comment (CTCaer) [inline bootloader/hos/secmon_exo.c]: You just broke 8.0.X support.

File: bootloader/sec/se.c [hunk 5]
```diff
@@ -167,6 +167,7 @@ int se_aes_unwrap_key(u32 ks_dst, u32 ks_src, const void *input)
 {
 	SE(SE_CONFIG_REG_OFFSET) = SE_CONFIG_DEC_ALG(ALG_AES_DEC) | SE_CONFIG_DST(DST_KEYTAB);
 	SE(SE_CRYPTO_REG_OFFSET) = SE_CRYPTO_KEY_INDEX(ks_src) | SE_CRYPTO_CORE_SEL(CORE_DECRYPT);
+	SE(0x318) = 0;
```
Comment (CTCaer) [inline bootloader/sec/se.c]: huh? Why?
it's already there. `SE(SE_BLOCK_COUNT_REG_OFFSET) = 0;`

**Pull request closed without merging**